app/src/main/java/ar/com/wolox/android/example/ui/home/news/NewsPresenter.kt

```diff
@@ -24,38 +24,97 @@ class NewsPresenter @Inject constructor(
 
     private var currentPage: Int = 1
     private var totalPages: Int = 1
+    private var activeCoroutine: Boolean = false
 
     private fun fetchNews() = launch {
-        networkRequest(newRepository.getNews(currentPage)) {
-            onResponseSuccessful { it ->
-                totalPages = it!!.totalPages
-                news = it.page
-                news.sortByDescending { newDate -> newDate.date }
-                currentPage++
+        if (!activeCoroutine) {
+            activeCoroutine = true
+            networkRequest(newRepository.getNews(currentPage)) {
+                onResponseSuccessful { it ->
+                    totalPages = it!!.totalPages
+                    news = it.page
+                    news.sortByDescending { newDate -> newDate.date }
+                    currentPage++
+                }
+                onResponseFailed { _, _ -> view?.showWrongCredentialsAlert() }
+                onCallFailure { view?.showNoNetworkAlert() }
             }
-            onResponseFailed { _, _ -> view?.showWrongCredentialsAlert() }
-            onCallFailure { view?.showNoNetworkAlert() }
+            view?.showNews(news)
+            activeCoroutine = false
```

Comment:
Move this to the line above, so the call to the view is made at the end of the method. 
Also, have you tested this concurrency when you enter the `onResponseFailed` method? What happens on that case? Maybe the `view?.showNews(news)` call can be in the `onResponseSuccessful()` method.